R/surv-roc_survival_curve.R

```diff
@@ -98,6 +98,26 @@ roc_curve_survival_impl <- function(truth,
   data <- dplyr::tibble(event_time, delta, estimate)
   data <- tidyr::unnest(data, cols = estimate)
 
+  .eval_times <- unique(data$.eval_time)
+
+  out <- list()
```

Comment:
should you pre-allocate the list? Maybe use `vector(mode = "list", length = length(.eval_times))`

Reply:
The way we are using it, I don't think it matters too much

``` r
data(ames, package = "modeldata")

f1 <- function(n) {
  out <- list()
  
  for (i in seq_len(n)) {
    out[[i]] <- ames
  }
  out
}

f2 <- function(n) {
  out <- vector(mode = "list", length = length(n))
  
  for (i in seq_len(n)) {
    out[[i]] <- ames
  }
  out
}

bench::mark(
  f1(5),
  f2(5)
)
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 f1(5)        4.26µs   4.39µs   217895.    7.14MB      0  
#> 2 f2(5)         4.3µs   4.51µs   216270.  160.56KB     21.6

bench::mark(
  f1(500),
  f2(500)
)
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 f1(500)       442µs    455µs     2177.    80.3KB     8.25
#> 2 f2(500)       442µs    456µs     2186.    80.3KB     6.14
```

<sup>Created on 2023-04-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
lists appear to not care about this as much as numerics do

``` r
f1 <- function(n) {
  out <- numeric()
  
  for (i in seq_len(n)) {
    out[[i]] <- i
  }
  out
}

f2 <- function(n) {
  out <- numeric(n)
  
  for (i in seq_len(n)) {
    out[[i]] <- i
  }
  out
}

bench::mark(
  f1(5),
  f2(5)
)
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 f1(5)         902ns    984ns   912421.    4.18MB       0 
#> 2 f2(5)         533ns    615ns  1516210.   26.53KB     152.

bench::mark(
  f1(500),
  f2(500)
)
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 f1(500)      50.6µs   57.2µs    16807.   80.28KB    67.1 
#> 2 f2(500)        11µs   11.4µs    85729.    3.95KB     8.57
```